### PR TITLE
chore: Remove unecessary filter

### PR DESCRIPTION
**Pull request by haseebnaseem**
## Problem
While working on issue (https://github.com/PostHog/posthog/issues/12391) I found that the filter `TermSearchFilterBackend` is not used for filtering at all. If removed, all the tests pass regardless.

In `TermSearchFilterBackend` there is a condition:
```
if isinstance(queryset, RawQuerySet):
            return queryset
```
There are two instances where TermFilterBackend is being used:

- In EventDefinitionViewSet in which the get_queryset function returns a raw query set. Hence branching into the conditional mentioned above
- In PropertyDefinitionViewSet in which the get_queryset function also returns a raw query set. Hence branching into the conditional mentioned above

## Changes

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

**Comment by benjackwhite** [comment]
I wouldn't assume that because we have no tests, it has no functionality 😉 we aim for 100% test coverage but don't necessarily have it. Will have to take a look and see who added it to make sure there isn't a hidden use case for this code

File: posthog/api/event_definition.py [hunk 1]
```diff
@@ -9,7 +9,7 @@
 from posthog.api.utils import create_event_definitions_sql
 from posthog.constants import AvailableFeature, EventDefinitionType
 from posthog.exceptions import EnterpriseFeatureException
-from posthog.filters import TermSearchFilterBackend, term_search_filter_sql
+from posthog.filters import term_search_filter_sql
```
Comment (benjackwhite) [inline posthog/api/event_definition.py:12]: Had a play myself and this indeed seems to do nothing. Could you also delete the `TermSearchFilterBackend` class and then we can get this in 💪 nice tidying!

**Review by benjackwhite — CHANGES_REQUESTED** [review]

**Comment by haseebnaseem** [comment]
> I wouldn't assume that because we have no tests, it has no functionality 😉 we aim for 100% test coverage but don't necessarily have it. Will have to take a look and see who added it to make sure there isn't a hidden use case for this code

@benjackwhite  Yes sorry I missed the second part of my explanation which is that in the filter we have a condition:
```
if isinstance(queryset, RawQuerySet):
            return queryset
```

There are two instances where `TermFilterBackend` is being used:
- In `EventDefinitionViewSet` in which the `get_queryset` function returns a raw query set. Hence branching into the conditional mentioned above
- In `PropertyDefinitionViewSet` in which the `get_queryset` function also returns a raw query set. Hence branching into the conditional mentioned above

File: posthog/api/event_definition.py (hunk 1 above)
Comment (haseebnaseem) [reply posthog/api/event_definition.py:12]: Removed in 9d54b4d

**Review by benjackwhite — APPROVED** [review]
Sweet - thanks for that!

**Comment by haseebnaseem** [comment]
@benjackwhite It seems that some of the tests are failing although they are unrelated to these changes. Is it possible to re-run the tests ?

**Comment by haseebnaseem** [comment]
@benjackwhite the CI seems to have failed again on the job [Django tests – FOSS (persons-on-events off), Py 3.8.14, clickhouse/clickhouse-server:22.3 (5/5)](https://github.com/PostHog/posthog/actions/runs/3641086493/jobs/6177707828#logs)
More specifically, the tests in file `posthog/clickhouse/test_schema.py` are failing. The PR doesn't really have anything to with the schema. Would appreciate a look and see how we can get this nice little cleanup merged.

**Pull request closed without merging**